polkadot/xcm/xcm-builder/src/weight.rs

```diff
@@ -257,3 +267,16 @@ impl<
 		OnUnbalanced::on_unbalanced(Currency::issue(self.1));
 	}
 }
+
+impl<WeightToFee, AssetId, AccountId, Currency, OnUnbalanced> Clone
+	for UsingComponents<WeightToFee, AssetId, AccountId, Currency, OnUnbalanced>
+where
+	WeightToFee: WeightToFeeT<Balance = Currency::Balance>,
+	AssetId: Get<MultiLocation>,
+	Currency: CurrencyT<AccountId>,
+	OnUnbalanced: OnUnbalancedT<Currency::NegativeImbalance>,
+{
+	fn clone(&self) -> UsingComponents<WeightToFee, AssetId, AccountId, Currency, OnUnbalanced> {
```

Comment:
Why is this custom impl needed? There is a `CloneNoBound` if the trait bounds are an issue.